pkg/apis/serving/v1/revision_lifecycle.go

```diff
@@ -226,7 +226,8 @@ func (rs *RevisionStatus) PropagateAutoscalerStatus(ps *autoscalingv1alpha1.PodA
 		// ScaleTargetInitialized down the road, we would have marked resources
 		// unavailable here, and have no way of recovering later.
 		// If the ResourcesAvailable is already false, don't override the message.
-		if !ps.IsScaleTargetInitialized() && !resUnavailable && ps.ServiceName != "" {
+		scaleTargetInitializedCondReason := ps.GetCondition(autoscalingv1alpha1.PodAutoscalerConditionScaleTargetInitialized).GetReason()
+		if !ps.IsScaleTargetInitialized() && !resUnavailable && ps.ServiceName != "" && scaleTargetInitializedCondReason != autoscalingv1alpha1.PendingReason {
```

Comment:
Can you clarify what is happening here? Generally we're not looking at the Reason string but Status.

To get here `PA::Ready=False` are we trying to set the right error message?

Another way to think about this can we just look at `ScaleTargetInitialized=False` to simplify this? Technically `not IsScaleTargetInitialized` is true when it is set to Unknown and False

Reply:
I think the new situation here is that the condition is now there on the creation of the resource with the generated defaulting, so I wasn't sure how else to differentiate between the default Unknown (with reason Pending) vs Unknown because something happened.

Reply:
> default Unknown (with reason Pending) vs Unknown because something happened.

I would view these two scenarios as being the same